code/game/objects/items/spear.dm

```diff
@@ -153,8 +153,14 @@
 	if(iseffect(AM)) //and no accidentally wasting your moment of glory on graffiti
 		return
 	user.say("[war_cry]", forced="spear warcry")
-	explosive.forceMove(AM)
-	explosive.detonate(lanced_by=user)
+	if(isliving(user))
+		var/mob/living/living_user = user
+		living_user.set_resting(TRUE, TRUE, TRUE)
+		living_user.Move(get_turf(AM))
+		explosive.forceMove(get_turf(living_user))
```

Comment:
if you're moving the explosive to the user's turf instead of whatever they attacked, you should remove the checks against trying to elance indestructible/godmoded targets